### ISSUE TITLE
Simpler patching mechanism

### ISSUE THREAD

**Issue by viovanov** [cloudfoundry-incubator/kubecf#604]
**Is your feature request related to a problem? Please describe.**
Easier editing/reading of patches.

**Describe the solution you'd like**
Instead of writing shell for patches, include the entire file:
- easier to edit
- no more shell
- no more diff conflicts

**Additional context**
The long term goal is to avoid patches.

**Comment by f0rmiga** [comment]
+1 on having a simpler patching mechanism.

The only problem with including the entire file is if the original file changes from one release to another, and we miss these changes. We can solve this problem by calculating the hash of the original file and then checking for it before we replace the file. When doing a CF bump, the patching would fail if the original file changed.

As you say, this shouldn't be too hard to fix given the fact we expect to move the fixes to the release original source codes.

**Comment by jandubois** [comment]
I agree that the patches are hard to read, but at least they provide the information about *what* has changed. Including the whole file hides that information; you will have to fetch the upstream source at the correct commit level and do a diff locally.

As @f0rmiga already wrote, it also no longer throws an error when there is a diff conflict. I think failing on these conflicts is a feature to alert us that something has changed upstream.

Using a hash of the upstream file will create even more conflicts because `patch` will apply the diff with "fuzz" if upstream changes did not directly modify the lines we are patching; the hash check will fail on any upstream change (maybe that is a feature, but it conflicts with the goal of "no more diff conflicts").

The long term goal of avoiding patches and getting them included upstream is the proper solution for this issue.

**Comment by f0rmiga** [comment]
The hash comparison forces the person who is bumping the release to re-validate the patch, i.e. to see if the changed parts are still relevant and/or play nice with the new release's incoming change(s).

**Comment by jandubois** [comment]
> The hash comparison forces the person who is bumping the release to re-validate the patch, i.e. to see if the changed parts are still relevant and/or play nice with the new release's incoming change(s).

Ok, that is fine, as it is more robust. It will just require even more frequent updating of the patches.

However updating the patch is now more error-prone, as you need to get the same file from the older release again to re-create the patch, and you can potentially make additional mistakes here. This change is just moving the ugliness from one part of the workflow to another.

I don't know, this just seems less transparent to me, as you can no longer see the changes in the patch with the Github website.
